src/oic/oic/consumer.py

```diff
@@ -496,7 +504,12 @@ def complete(self, state, authn_method: str = "client_secret_basic"):
         if resp.type() == "ErrorResponse":
             raise TokenError(resp.error, resp)
 
-        self._backup(state)
+        if session_state:
```

Comment:
```suggestion
        if session_state is not None:
```
explicit comparison is better.